network/src/commonTest/kotlin/com/wire/kalium/api/v0/message/QualifiedMessageApiV0Test.kt

```diff
@@ -0,0 +1,126 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.api.v0.message
+
+import com.wire.kalium.api.ApiTest
+import com.wire.kalium.api.json.ValidJsonProvider
+import com.wire.kalium.api.tools.IgnoreIOS
+import com.wire.kalium.model.QualifiedSendMessageRequestJson
+import com.wire.kalium.model.QualifiedSendMessageResponseJson
+import com.wire.kalium.network.api.base.authenticated.message.EnvelopeProtoMapperImpl
+import com.wire.kalium.network.api.base.authenticated.message.MessageApi
+import com.wire.kalium.network.api.base.authenticated.message.QualifiedSendMessageResponse
+import com.wire.kalium.network.api.base.model.ConversationId
+import com.wire.kalium.network.api.v0.authenticated.MessageApiV0
+import com.wire.kalium.network.exceptions.ProteusClientsChangedError
+import com.wire.kalium.network.utils.isSuccessful
+import io.ktor.http.HttpStatusCode
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+@ExperimentalCoroutinesApi
+@IgnoreIOS
+class QualifiedMessageApiV0Test : ApiTest {
+
+    @Test
+    fun givenAValid_whenSendingAMessage_theRequestShouldBeConfiguredCorrectly() =
+        runTest {
+            val networkClient = mockAuthenticatedNetworkClient(
+                SUCCESS_RESPONSE.rawJson,
+                statusCode = HttpStatusCode.OK,
+                assertion =
+                {
+                    assertPost()
+                    assertXProtobuf()
+                    assertPathEqual(SEND_MESSAGE_PATH)
+                }
+            )
+
+            val messageApi: MessageApi = MessageApiV0(networkClient, EnvelopeProtoMapperImpl())
+            val response = messageApi.qualifiedSendMessage(
+                DEFAULT_PARAMETERS_RESPONSE.serializableData,
+                TEST_CONVERSATION_ID
+            )
+
+            assertTrue(response.isSuccessful())
+            assertEquals(response.value, SUCCESS_RESPONSE.serializableData)
+        }
+
+    @Test
+    fun givenDeletedUsersError_whenSendingAMessage_TheCorrectErrorIsPropagate() = errorCaseTest(
+        DELETED_ERROR_RESPONSE
+    )
+
+    @Test
+    fun givenRedundantUsersError_whenSendingAMessage_TheCorrectErrorIsPropagate() = errorCaseTest(
+        REDUNDANT_ERROR_RESPONSE
+    )
+
+    @Test
+    fun givenMissingUsersError_whenSendingAMessage_TheCorrectErrorIsPropagate() = errorCaseTest(
+        MISSING_ERROR_RESPONSE
+    )
+
+    @Test
+    fun givenFailedToSentUsersError_whenSendingAMessage_TheCorrectErrorIsPropagate() = errorCaseTest(
```

Comment:
I suppose this will only be available on API v4 when it gets implemented?

Reply:
Confirmed with Marco and should be v3 and up ! so I made the changes